src/code/z_demo.c

```diff
@@ -971,144 +1003,145 @@ s32 Cutscene_CountNormalMasks(void) {
+                csCtx->curFrame--;
             }
-            D_801BB160 = CS_TEXTBOX_TYPE_1;
-            D_801BB124 = cmd->base;
-            if (cmd->type == CS_TEXTBOX_TYPE_BOSSES_REMAINS) {
+            sCutsceneTextboxType = CS_TEXT_TYPE_1;
+            sCurTextId = cmd->textId;
+            if (cmd->type == CS_TEXT_TYPE_BOSSES_REMAINS) {
                 if (CHECK_QUEST_ITEM(QUEST_REMAINS_ODOLWA) && CHECK_QUEST_ITEM(QUEST_REMAINS_GOHT) &&
                     CHECK_QUEST_ITEM(QUEST_REMAINS_GYORG) && CHECK_QUEST_ITEM(QUEST_REMAINS_TWINMOLD)) {
-                    if (cmd->textId1 != 0xFFFF) {
-                        Message_StartTextbox(play, cmd->textId1, NULL);
+                    if (cmd->altTextId1 != 0xFFFF) {
+                        Message_StartTextbox(play, cmd->altTextId1, NULL);
                     }
                 } else {
-                    Message_StartTextbox(play, cmd->base, NULL);
+                    Message_StartTextbox(play, cmd->textId, NULL);
                 }
-            } else if (cmd->type == CS_TEXTBOX_TYPE_ALL_NORMAL_MASKS) {
+            } else if (cmd->type == CS_TEXT_TYPE_ALL_NORMAL_MASKS) {
                 if (Cutscene_CountNormalMasks() == 20) {
-                    if (cmd->textId1 != 0xFFFF) {
-                        Message_StartTextbox(play, cmd->textId1, NULL);
+                    if (cmd->altTextId1 != 0xFFFF) {
+                        Message_StartTextbox(play, cmd->altTextId1, NULL);
                     }
                 } else {
-                    Message_StartTextbox(play, cmd->base, NULL);
+                    Message_StartTextbox(play, cmd->textId, NULL);
                 }
             } else {
-                Message_StartTextbox(play, cmd->base, NULL);
+                Message_StartTextbox(play, cmd->textId, NULL);
             }
-        } else {
-            goto else_label;
+            goto end;
         }
-    } else if (D_801BB128 != cmd->base) {
-        D_801BB160 = CS_TEXTBOX_TYPE_LEARN_SONG;
-        D_801BB128 = cmd->base;
-        func_80152434(play, cmd->base);
     } else {
-    else_label:
-        if (csCtx->frames >= cmd->endFrame) {
-            // The Textbox command can change the current cutscene frame, mainly to prevent advancing the cutscene when
-            // a textbox that is expected to be closed by the user is still open.
-
-            originalCsFrames = csCtx->frames;
-            talkState = Message_GetState(&play->msgCtx);
-            if ((talkState != TEXT_STATE_CLOSING) && (talkState != TEXT_STATE_NONE) && (talkState != TEXT_STATE_7) &&
-                (talkState != TEXT_STATE_8)) {
-                csCtx->frames--;
-                if ((talkState == TEXT_STATE_CHOICE) && Message_ShouldAdvance(play)) {
-                    if (play->msgCtx.choiceIndex == 0) {
-                        if (cmd->base == 0x33BD) {
-                            func_8019F230();
-                        }
+        if (sCurOcarinaAction != cmd->textId) {
```

Comment:
What do you think about merging this `else { if` into an `else if`?

Reply:
I intentionally left it as `else {` so that the different levels of `if`'s line up. i.e.:
```c
if (cmd->type != CS_TEXT_OCARINA_ACTION) {
    if (sCurTextId != cmd->textId) {
        ....
} else {
    if (sCurOcarinaAction != cmd->textId) {
        ...
```
That second level of "ifs" are checking if this is a new textId or if it's a continuing one

Reply:
somewhat related, but the if statements should be flipped so that they start with `cmd->textId ==`

Reply:
My preference is to have the variables aligned, because it's checked and set as:
```c
if (sCurTextId != cmd->textId) {
    sCurTextId = cmd->textId;
```
and
```c
if (sCurOcarinaAction != cmd->textId) {
    sCurOcarinaAction = cmd->textId;
```